src/persistence/database_manager.py

```
@@ -272,6 +341,67 @@ def update_novel_active_plot(self, novel_id: int, plot_id: Optional[int]):
+        novel = self.get_novel_by_id(novel_id)
+        if not novel:
+            raise ValueError(f"Novel with ID {novel_id} not found.")
+
+        if novel.get('active_plot_id') is not None:
+            # Verify the plot actually exists, otherwise it's a dangling ID
+            plot_check = self.get_plot_by_id(novel['active_plot_id'])
+            if plot_check:
+                return novel['active_plot_id']
+            else:
+                # Dangling active_plot_id, proceed to create a new one
+                print(f"Warning: Novel {novel_id} had a dangling active_plot_id {novel['active_plot_id']}. Creating a new plot.")
```

Pylint (3.3.7)

[refactor] Unnecessary "else" after "return", remove the "else" and de-indent the code inside it

(R1705)